include/soci/soci-types.h

```diff
@@ -6,7 +6,7 @@
 #if defined(__GNUC__) || defined(__clang__)
     #if defined(__LP64__)
         #define SOCI_LONG_IS_64_BIT 1
-        #if SOCI_OS == SOCI_OS_LINUX || SOCI_OS == SOCI_OS_FREE_BSD
+        #if SOCI_OS == SOCI_OS_LINUX || SOCI_OS == SOCI_OS_FREE_BSD || SOCI_OS == SOCI_OS_SUN
```

Comment:
Here too we should just check for the condition directly instead of testing for the platform... In fact, I really don't know why did we do it like this in the first place, @zann1x do you remember if there was some good reason for it?

Reply:
I don't think there was any other reason for it than to just make it work both locally and in the CI pipeline. In the end, it's only important that we know which data type definitions the underlying OS uses.